docs/DevGuide/ImplementingVectors.md

```diff
@@ -45,13 +45,29 @@ DataVector>`. This template pattern is known as the
 ["Curiously Recurring Template Pattern"]
 (https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern) (CRTP).
 
+For the Blaze system to use the CRTP inheritance appropriately, it requires the
+specification of separate type traits in the `blaze` namespace.
+These traits can usually be declared in a standard form, so are abstracted in a
+macro.
+For any new vector `MyNewVector`, the pattern that must appear at the beginning
+of the file (i.e. before the class definition) is:
+```
+/// \cond
+class MyNewVector;
+/// \endcond
+namespace blaze {
+DECLARE_GENERAL_VECTOR_BLAZE_TRAITS(MyNewVector);
+}
+```
+
 The class template `VectorImpl` defines various constructors, assignment
-operators, and iterator generation members. Most of these are inherited from
-Blaze types, but in addition, the methods `set_data_ref`, and `pup` are defined
-for use in SpECTRE. All except for the assignment operators and constructors
-will be implicitly inherited from `VectorImpl`. The assignment and constructors
-may be inherited calling the following alias code in the vector class
-definition:
+operators, and iterator generation members.
+Most of these are inherited from Blaze types, but in addition, the methods
+`set_data_ref`, and `pup` are defined for use in SpECTRE.
+All except for the assignment operators and constructors will be implicitly
+inherited from `VectorImpl`.
+The assignment and constructors may be inherited calling the following alias
+code in the vector class definition:
```

Comment:
You've done some weird formatting adding lots of newlines in this file, and it's caused strange formatting in the rendered version in some places (the last code bit in the last sentence on the page, for example).